src/components/date_time.rs

```diff
@@ -148,3 +148,162 @@ impl From<NaiveDate> for DatePerhapsTime {
         Self::Date(date)
     }
 }
+
+#[cfg(feature = "parser")]
+impl TryFrom<crate::parser::Property<'_>> for DatePerhapsTime {
+    type Error = &'static str;
+
+    fn try_from(value: crate::parser::Property) -> Result<Self, Self::Error> {
+        let val = value.val.as_ref();
+
+        // UTC is here first because lots of fields MUST be UTC, so it should,
+        // in practice, be more common that others.
+        if let Ok(utc_dt) = Utc.datetime_from_str(val, "%Y%m%dT%H%M%SZ") {
+            return Ok(Self::DateTime(CalendarDateTime::Utc(utc_dt)));
+        };
+
+        if let Ok(naive_date) = NaiveDate::parse_from_str(val, "%Y%m%d") {
+            return Ok(Self::Date(naive_date));
+        };
+
+        if let Ok(naive_dt) = NaiveDateTime::parse_from_str(val, "%Y%m%dT%H%M%S") {
+            if let Some(tz_param) = value.params.iter().find(|p| p.key == "TZID") {
+                if let Some(tzid) = &tz_param.val {
+                    return Ok(Self::DateTime(CalendarDateTime::WithTimezone {
+                        date_time: naive_dt,
+                        tzid: tzid.as_ref().to_string(),
+                    }));
+                } else {
+                    return Err("Found empty TZID param.");
```

Comment:
This function returns two distinct `Err(&str)`. I'm not sure if a custom error type is preferring. The boilerplate sounds a bit much for JUST this function.